src/java/main/org/apache/zookeeper/server/NIOServerCnxnFactory.java

```diff
@@ -842,7 +868,26 @@ private void addCnxn(NIOServerCnxn cnxn) {
 
     protected NIOServerCnxn createConnection(SocketChannel sock,
             SelectionKey sk, SelectorThread selectorThread) throws IOException {
-        return new NIOServerCnxn(zkServer, sock, sk, this, selectorThread);
+
+        NIOServerCnxn cnxn = null;
+
+        if (serverCnxnClassCtr != null) {
```

Comment:
nothing is wrong with it per say - just overhead, as we will call "newInstance" on the constructor for every code path - test or not.